pkg/syncer/metadatasyncer.go

```diff
@@ -931,8 +971,9 @@ func csiPVDeleted(ctx context.Context, pv *v1.PersistentVolume, metadataSyncer *
 			deleteDisk = true
 		}
 		log.Debugf("PVDeleted: vSphere CSI Driver is deleting volume %v with delete disk %v", pv, deleteDisk)
-		if err := metadataSyncer.volumeManager.DeleteVolume(ctx, pv.Spec.CSI.VolumeHandle, deleteDisk); err != nil {
-			log.Errorf("PVDeleted: Failed to delete disk %s with error %+v", pv.Spec.CSI.VolumeHandle, err)
+
+		if err := metadataSyncer.volumeManager.DeleteVolume(ctx, volumeHandle, deleteDisk); err != nil {
```

Comment:
if the Migration feature flag is disabled and we deploy driver with this change, and PV we are dealing with is migrated PV, `volumeHandle` will be the empty string.

Can we make sure to check `volumeHandle` is not empty string before calling vSphere APIs.

Do this for all cases.

Reply:
> 
> Can we make sure to check `volumeHandle` is not empty string before calling vSphere APIs.
> 
> Do this for all cases.

This is handled in all the calls (pvUpdated, pvDeleted, pvcUpdated & pvcDeleted).
See : https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/264/files#diff-6fc246216efb3eb9047f82d1116e9813R615

I don't think we need the volumeHandle == "" check.

Reply:
Can you update the PR for the case we discussed?

Reply:
Done

Reply:
<pre>
2020-07-16T00:19:45.853Z	WARN	syncer/metadatasyncer.go:584	volume-migration feature switch is disabled. Cannot update vSphere volume pvc-618dd316-50ec-4a08-8542-58810c972814	{"TraceId": "ed68ed89-0595-422b-9a63-6efb0e12f706"}
</pre>